jellyfin_kodi/views.py

```diff
@@ -105,29 +105,56 @@
 
 
 def verify_kodi_defaults():
-
     ''' Make sure we have the kodi default folder in place.
     '''
-    node_path = xbmc.translatePath("special://profile/library/video")
-
-    if not os.path.exists(node_path):
-        try:
-            shutil.copytree(
-                src=xbmc.translatePath("special://xbmc/system/library/video"),
-                dst=xbmc.translatePath("special://profile/library/video"))
-        except Exception as error:
-            LOG.warning(error)
-            xbmcvfs.mkdir(node_path)
 
+    source_base_path = xbmc.translatePath("special://xbmc/system/library/video")
+    dest_base_path = xbmc.translatePath("special://profile/library/video")
+
+    # Make sure the files exist in the local profile.
+    # TODO: Investigate why this is needed.
+    # I would think Kodi pulls data from the default profile
+    # if we don't do this.
+    for source_path, dirs, files in os.walk(source_base_path):
+        relative_path = os.path.relpath(source_path, source_base_path)
+        dest_path = os.path.join(dest_base_path, relative_path)
+
+        if not os.path.exists(dest_path):
+            os.mkdir(dest_path)
+
+        for file_name in files:
+            dest_file = os.path.join(dest_path, file_name)
+            copy = False
+
+            if not os.path.exists(dest_file):
+                copy = True
+            elif os.path.splitext(file_name)[1].lower() == '.xml':
+                try:
+                    etree.parse(dest_file)
+                except etree.ParseError:
+                    LOG.warning("Unable to parse `{}`, recovering from default.".format(dest_file))
+                    copy = True
+
+            if copy:
+                source_file = os.path.join(source_path, file_name)
+                LOG.debug("Copying `{}` -> `{}`".format(source_file, dest_file))
+                xbmcvfs.copy(source_file, dest_file)
+
+    # This code seems to enforce a fixed ordering.
+    # Is it really desirable to force this on users?
+    # The default (system wide) order is [10, 20, 30] in Kodi 19.
```

Comment:
Best I can tell, this ordering isn't actually doing anything beneficial.  Especially since these are all localized in each library type, it seems largely pointless.  My only concern with changing it is that we'd probably need to write a migration so people don't get double entries (haven't tested, just my assumption)